File: killrvideo-service-videocatalog/src/main/java/com/killrvideo/service/video/dao/UserVideoDao.java
package com.killrvideo.service.video.dao;

import com.datastax.oss.driver.api.mapper.annotations.Dao;
import com.datastax.oss.driver.api.mapper.annotations.Insert;
import com.datastax.oss.driver.api.mapper.annotations.StatementAttributes;
import com.killrvideo.service.video.dto.UserVideo;

import java.util.concurrent.CompletableFuture;

@Dao
public interface UserVideoDao {
    @Insert
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    CompletableFuture<Void> insert(UserVideo userVideo);
}

File: killrvideo-service-videocatalog/src/main/kotlin/com/killrvideo/service/video/dao/UserVideoDao.kt
package com.killrvideo.service.video.dao

import com.datastax.oss.driver.api.mapper.annotations.Dao
import com.datastax.oss.driver.api.mapper.annotations.Insert
import com.datastax.oss.driver.api.mapper.annotations.StatementAttributes
import com.killrvideo.service.video.dto.UserVideo
import java.util.concurrent.CompletableFuture

@Dao
interface UserVideoDao {
    @Insert
    @StatementAttributes(consistencyLevel = "LOCAL_QUORUM")
    fun insert(userVideo: UserVideo): CompletableFuture<Void>
}
